peekingduck/pipeline/nodes/dabble/utils/tracking_files/iou_tracker/track.py

```diff
@@ -0,0 +1,176 @@
+
+        if data_output_format == 'mot_challenge':
+            self.output = self.get_mot_challenge_format
+        else:
+            raise NotImplementedError
+
+    # pylint: disable=too-many-arguments
+    def update(self,
+               frame_id: int,
+               bbox: np.ndarray,
+               detection_confidence: float,
+               class_id: Union[str, int] = None,
+               lost: int = 0,
+               iou_score: float = 0., **kwargs: Any) -> None:
+        """
+        Update the track.
+
+        Args:
+            frame_id (int): Camera frame id.
+            bbox (numpy.ndarray): Bounding box pixel coordinates as
+                (xmin, ymin, width, height) of the track.
+            detection_confidence (float): Detection confidence of the object (probability).
+            class_id (int or str): Class label id.
+            lost (int): Number of times the object or track was not tracked
+                by tracker in consecutive frames.
+            iou_score (float): Intersection over union score.
+            kwargs (dict): Additional key word arguments.
+        """
+        self.class_id = class_id
+        self.bbox = np.array(bbox)
+        self.detection_confidence = detection_confidence
+        self.frame_id = frame_id
+        self.iou_score = iou_score
+
+        if lost == 0:
+            self.lost = 0
+        else:
+            self.lost += lost
+
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+        self.detection_confidence_max = max(self.detection_confidence_max,
+                                            detection_confidence)
+
+        self.age += 1
+
+    @property
+    def centroid(self) -> np.ndarray:
+        """
+        Return the centroid of the bounding box.
+
+        Returns:
+            numpy.ndarray: Centroid (x, y) of bounding box.
+
+        """
+        return np.array((self.bbox[0]+0.5*self.bbox[2], self.bbox[1]+0.5*self.bbox[3]))
+
+    def get_mot_challenge_format(self) -> \
+                Tuple[int, int, Any, Any, Any, Any, float, int, int, int]:
+        """
+        Get the tracker data in MOT challenge format as a tuple of elements containing
+        `(frame, id, bb_left, bb_top, bb_width, bb_height, conf, x, y, z)`
+
+        References:
+            - Website : https://motchallenge.net/
+
+        Returns:
+            tuple: Tuple of 10 elements representing `(frame, id, bb_left,
+                    bb_top, bb_width, bb_height, conf, x, y, z)`.
+
+        """
+        mot_tuple = (
+            self.frame_id, self.id_num, self.bbox[0], self.bbox[1], self.bbox[2],
+            self.bbox[3], self.detection_confidence, -1, -1, -1
+        )
+        return mot_tuple
+
+    # pylint: disable=no-self-use
+    def predict(self) -> None:
+        """Implement to prediction the next estimate of track."""
+        # pylint: disable=notimplemented-raised
+        # pylint: disable=raising-bad-type
+        raise NotImplemented
```

Comment:
Why not just remove this if it is not used and not implemented?

Reply:
Function removed